src/nvim/api/autocmd.c

```diff
@@ -763,10 +782,119 @@ static bool unpack_string_or_array(Array *array, Object *v, char *k, Error *err)
     }
     *array = copy_array(v->data.array);
   } else {
-    api_set_error(err,
-                  kErrorTypeValidation,
-                  "'%s' must be an array or a string.",
-                  k);
+    if (required) {
+      api_set_error(err,
+                    kErrorTypeValidation,
+                    "'%s' must be an array or a string.",
+                    k);
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// Returns AUGROUP_ERROR if there was a problem with {group}
+static int get_augroup_from_object(Object group, Error *err)
```

Comment:
I think this could be used in two more functions: [`nvim_get_autocmds`](https://github.com/neovim/neovim/blob/d8bf4b413574c4b12526b97ee73445e026a7ea29/src/nvim/api/autocmd.c#L65-L86) and [`nvim_do_autocmd`](https://github.com/neovim/neovim/blob/d8bf4b413574c4b12526b97ee73445e026a7ea29/src/nvim/api/autocmd.c#L680-L703) in a similar way as done in `nvim_create_autocmd`.

Reply:
Sounds like a good refactor for a follow-up PR.